tests/vm_test.go

```diff
@@ -289,15 +289,18 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 			vm := createVM(virtClient, libvmifact.NewCirros())
 
-			err = retryWithMetadataIfModified(vm.ObjectMeta, func(meta k8smetav1.ObjectMeta) error {
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Get(context.Background(), meta.Name, k8smetav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				vm.Spec.Template.ObjectMeta.Annotations = annotations
-				vm, err = virtClient.VirtualMachine(meta.Namespace).Update(context.Background(), vm, metav1.UpdateOptions{})
-				return err
-			})
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, k8smetav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			annotationsPatch, err := patch.New(
+				patch.WithAdd(
+					"/spec/template/spec/metadata/annotations",
+					annotations,
+				)).GeneratePayload()
 			Expect(err).ToNot(HaveOccurred())
 
+			vm, err = virtClient.VirtualMachine(vm.Namespace).Patch(context.Background(), vm.Name, types.JSONPatchType, annotationsPatch, metav1.PatchOptions{})
+
```

Comment:
nit: The `Expect` fits immediately after the operation without a newline.
The newline is after the `Expect` to separate it from the next operation of starting the VM.